### PR TITLE
Support new sdk flag `--primary-network-endpoint`

### PR DESCRIPTION
**Pull request by icezohu**

### CODE REVIEW

File: apps/indexer-coordinator/src/project/project.service.ts [hunk 1]
```diff
@@ -174,6 +174,9 @@ export class ProjectService {
       project = await this.addProject(id);
     }
 
+    baseConfig.usePrimaryNetworkEndpoint =
+      baseConfig.usePrimaryNetworkEndpoint === false ? false : true;
```
Comment (ianhe8x) [inline apps/indexer-coordinator/src/project/project.service.ts:178]: Can this be rewritten in a more simple way?

Comment (icezohu) [reply apps/indexer-coordinator/src/project/project.service.ts]: readability improved. please review.

File: apps/indexer-coordinator/src/project/project.service.ts [hunk 2]
```diff
@@ -215,6 +219,10 @@ export class ProjectService {
 
     const mmrPath = argv['mmrPath'].replace(/\/$/, '');
 
+    if (baseConfig.usePrimaryNetworkEndpoint !== false) {
```
Comment (mzxyz) [inline apps/indexer-coordinator/src/project/project.service.ts:222]: [suggestion] can be a share function in this service as there has 2 places have same code

Comment (ianhe8x) [inline apps/indexer-coordinator/src/project/project.service.ts:222]:
```suggestion
    if (baseConfig.usePrimaryNetworkEndpoint === undefined) {
```

**Review by mzxyz — APPROVED** [review]

**Pull request merged**